python/ray/tune/tests/mock.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import sys
+
+from ray.rllib.agents.mock import _MockTrainer
+from ray.tune import DurableTrainable
+from ray.tune.sync_client import get_sync_client
+from ray.tune.syncer import NodeSyncer
+
+if sys.version_info >= (3, 3):
+    from unittest.mock import patch
+else:
+    from mock import patch
+
+MOCK_REMOTE_DIR = "/tmp/mock-tune-remote/"
+# Sync and delete templates that operate on local directories.
+LOCAL_SYNC_TEMPLATE = "mkdir -p {target} && rsync -avz {source}/ {target}/"
+LOCAL_DELETE_TEMPLATE = "rm -rf {target}"
+
+
+def mock_storage_client():
+    """Mocks storage client that treats a local dir as durable storage."""
+    return get_sync_client(LOCAL_SYNC_TEMPLATE, LOCAL_DELETE_TEMPLATE)
+
+
+class MockNodeSyncer(NodeSyncer):
+    """Mock NodeSyncer that syncs to and from /tmp"""
+
+    def has_remote_target(self):
+        return True
+
+    @property
+    def _remote_path(self):
+        if self._remote_dir.startswith('/'):
+            self._remote_dir = self._remote_dir[1:]
+        return os.path.join(MOCK_REMOTE_DIR, self._remote_dir)
+
+
+class MockRemoteTrainer(_MockTrainer):
+    """Mock Trainable that saves at tmp for simulated clusters."""
+
+    def __init__(self, *args, **kwargs):
+        super(MockRemoteTrainer, self).__init__(*args, **kwargs)
+        if self._logdir.startswith('/'):
```

Comment:
why would logdir start with `/`?

Reply:
as in its an absolute path, so you need to make it relative before prepending `/tmp/...`